src/conda_project/_conda_lock.py

```diff
@@ -25,6 +25,21 @@ def is_conda_lock_3() -> bool:
     from conda_lock.lookup import DEFAULT_MAPPING_URL  # pragma: no cover
 
 
+def get_virtual_package_spec(verbose: bool = False) -> Optional[Path]:
```

Comment:
could we move this in the Project class such that the path passed into the the call to _make_lock_files uses the project directory? This would allow the project to own its virtual-packages.yml file and be able to operate the project where the shell working dir is not the project dir (the `--directory`) flag

Reply:
Yes